src/libraries/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -176,80 +179,107 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
+            Assert.Throws<UnauthorizedAccessException>(() =>  CreateDirectoryWithSecurity(dirInfo, security));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_NotFound_FullControl()
+        {
+            using var tempRootDir = new TempAclDirectory();
+            string dirPath = Path.Combine(tempRootDir.Path, Guid.NewGuid().ToString(), "ParentDoesNotExist");
 
-            Assert.Throws<UnauthorizedAccessException>(() =>
+            var dirInfo = new DirectoryInfo(dirPath);
+            var security = GetDirectorySecurity(FileSystemRights.FullControl);
+            // Succeeds because it creates the missing parent folder
+            CreateDirectoryWithSecurity(dirInfo, security);
+        }
+
+        private void CreateDirectoryWithSecurity(DirectoryInfo info, DirectorySecurity security)
+        {
+            if (PlatformDetection.IsNetFramework)
             {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, security);
-                }
-                else
-                {
-                    info.Create(security);
-                }
-            });
+                FileSystemAclExtensions.Create(info, security);
+            }
+            else
+            {
+                info.Create(security);
+            }
         }
 
         [Fact]
         public void DirectoryInfo_Create_DefaultDirectorySecurity()
         {
-            DirectorySecurity security = new DirectorySecurity();
-            Verify_DirectoryInfo_Create(security);
+            var security = new DirectorySecurity();
+            Verify_DirectorySecurity_CreateDirectory(security);
         }
 
         [Theory]
-        [InlineData(FileSystemRights.ReadAndExecute, AccessControlType.Allow)]
-        [InlineData(FileSystemRights.ReadAndExecute, AccessControlType.Deny)]
-        [InlineData(FileSystemRights.WriteData,      AccessControlType.Allow)]
-        [InlineData(FileSystemRights.WriteData,      AccessControlType.Deny)]
-        [InlineData(FileSystemRights.FullControl,    AccessControlType.Allow)]
-        [InlineData(FileSystemRights.FullControl,    AccessControlType.Deny)]
-        public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(
-            FileSystemRights rights,
-            AccessControlType controlType)
+        // Must have at least one Read, otherwise the TempAclDirectory will fail to delete that item on dispose
+        [InlineData(FileSystemRights.FullControl)]
+        [InlineData(FileSystemRights.Read)]
+        [InlineData(FileSystemRights.Read | FileSystemRights.Write)]
+        [InlineData(FileSystemRights.Read | FileSystemRights.Write | FileSystemRights.ExecuteFile)]
+        [InlineData(FileSystemRights.ReadAndExecute)]
+        public void DirectoryInfo_Create_DirectorySecurity_SpecificAccessRule(FileSystemRights rights)
```

Comment:
```suggestion
        public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(FileSystemRights rights)
```
What's the motivation for changing the name?

Reply:
To match the other one.